src/mds/MDCache.cc

```diff
@@ -9713,10 +9713,18 @@ void MDCache::handle_discover(MDiscover *dis)
 
   assert(from != whoami);
 
+  if (mds->get_state() <= MDSMap::STATE_STARTING) {
+    dout(0) << "handle_discover: not started yet, delaying" << dendl;
+    mds->wait_for_mdsmap(mds->mdsmap->get_epoch(),
```

Comment:
What if we are not bound for STATE_ACTIVE? Don't we intend to drop the message under some circumstances on the next line?